Add FAQ entry for exporter failure (#448)
signalfx/splunk-otel-java: docs/faq.md

- **How often do you release?** We strive to release the Splunk distribution within 2 working days after
  the [upstream project](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases)
  releases. OpenTelemetry Java generally releases a new version every 4 weeks.
- **What happens when the exporter endpoint is temporarily unreachable?** Sending telemetry resumes when the endpoint

>   becomes available again. Telemetry collected in the meantime is dropped. To avoid dropping data, use the

Reviewer: The `BatchSpanProcessor` will buffer spans for a little bit, so maybe this is a bit vague. Not sure that it matters to capture much more detail, but it's not completely accurate, right?

>   [Splunk OpenTelemetry Connector](https://github.com/signalfx/splunk-otel-collector), which supports retrying failed

Reviewer 2: Wait, this claim about Connector is totally misleading. We have "JVM -> Collector -> Backend".  This FAQ item speaks about "what happens if Collector is down", meaning broken "JVM -> Collector" leg. Connector/Collector retrying works for broken "Collector -> Backend" leg.

Author: Removed collector part, is it ok now?

> - **What happens when exporter endpoint is temporarily unreachable?** Sending telemetry will resume when endpoint
>   becomes available again. Telemetry collected meanwhile will get dropped. To avoid dropping data it is recommended to
>   use [Splunk OpenTelemetry Connector](https://github.com/signalfx/splunk-otel-collector) that supports retrying failed
>   transmissions.

Reviewer 3: ```suggestion
- **What happens when the exporter endpoint is temporarily unreachable?** Sending telemetry resumes when the endpoint becomes available again. Telemetry collected in the meantime is dropped. To avoid dropping data, 
  use the [Splunk OpenTelemetry Connector](https://github.com/signalfx/splunk-otel-collector), which supports retrying failed transmissions.
```
